### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by Bond-009**
#2149

### CODE REVIEW

**Comment by barronpm** [comment]
This will have lots of conflicts with #3812 due to it touching API services

File: MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs [hunk 1]
```diff
@@ -163,17 +164,36 @@ private async Task<List<RemoteSearchResult>> GetSearchResultsGeneric(string name
...
+                throw new ArgumentException("String can't be null or empty.", nameof(name));
             }
 
-            var url3 = string.Format(_searchURL, WebUtility.UrlEncode(name), TmdbUtils.ApiKey, language, type);
+            string url3;
+            if (year != null && string.Equals(type, "movie", StringComparison.OrdinalIgnoreCase))
+            {
+                url3 = string.Format(
+                    CultureInfo.InvariantCulture,
+                    SearchUrl,
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs:175]:
```suggestion
                    SearchUrlWithYear ,
```

File: MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs [hunk 2]
```diff
@@ -163,17 +164,36 @@ private async Task<List<RemoteSearchResult>> GetSearchResultsGeneric(string name
...
+                    WebUtility.UrlEncode(name),
+                    TmdbUtils.ApiKey,
+                    language,
+                    type) + "&primary_release_year=" + year;
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs:179]:
```suggestion
                url3 = string.Format(
                    CultureInfo.InvariantCulture,
                    SearchUrlWithYear,
                    WebUtility.UrlEncode(name),
                    TmdbUtils.ApiKey,
                    language,
                    type,
                    year);
```

Comment (Bond-009) [reply MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs:179]: `first_air_date_year` != `primary_release_year`

Comment (crobibero) [reply MediaBrowser.Providers/Plugins/Tmdb/Movies/TmdbSearch.cs:179]: Read it too quickly! Then there should be another SearchUrl?

File: MediaBrowser.Controller/Entities/Movies/Movie.cs [hunk 3]
```diff
@@ -1,5 +1,7 @@
+
```
Comment (barronpm) [inline MediaBrowser.Controller/Entities/Movies/Movie.cs:1]: Why?

**Review by barronpm — APPROVED** [review]

**Pull request merged**